onnx/defs/traditionalml/defs.cc

```diff
@@ -355,6 +355,124 @@ ONNX_ML_OPERATOR_SET_SCHEMA(
+    and 'classes_strings' is ["Sally", "Dori", "Amy"]. The output would be
+    ["Dori", "Amy", "Amy", "Sally"]. The other mode maps non-integer types
+    to integer types, called loop-up mode. Each element in the input
+    tensor is used as a position index to the associated 'classes_*'
+    attribute. Let ["A", "B", "B", "C"] be the input, 'classes_strings' be
+    ["C", "B"], 'default_int64' be -99. Then, the corresponding output
+    should be [-99, 1, 1, 0] because "C" and "B" are respectively the first
+    (indexed by 0) and the second (indexed by 1) elements in
+    'classes_strings' while "A" cannot be found in 'classes_strings'.<br>
+    Under loop-up mode, only default_int64 can be set because it is the
+    integer value used whenever indexing fails (e.g., when an integer
+    larger than the size of 'classes_* is given). In the meanwhile, input
+    type is identical to the element type of the specified 'classes_' while
+    output is always an integer tensor. For indexing mode, the input type
+    is always integer tensor and the output type is identical to the
+    specified specified 'classes_*'. The 'default_*' field is the output
+    value whenever loop-up fails for an element. Note that only one of
+    'classes_*' fields can be set at one time.<br>
+    The input and output shapes of Label Encoder are the same. The output
+    element's type is determined by the specified 'classes_*' field. Notice
+    that only one 'classes_*' should be set at one time.<br>
+    For look-up, bit-wise comparision is used so even a float NaN can be
+    mapped to an integer.
+)DOC";
+
+ONNX_ML_OPERATOR_SET_SCHEMA(
+    LabelEncoder,
+    2,
+    OpSchema()
+        .SetDoc(LabelEncoder_ver2_doc)
+        .Input(0, "X", "Input data. It can be either tensor or scalar.", "T1")
+        .Output(0, "Y", "Output data. Under look-up mode, the output values are integers. Under indexing mode, the output type is determined by the specified 'classes_*'.", "T2")
+        .TypeConstraint(
+            "T1",
+            {"tensor(string)", "tensor(int64)", "tensor(float)"},
+            "The input type is a tensor of any shape.")
+        .TypeConstraint(
+            "T2",
+            {"tensor(string)", "tensor(int64)", "tensor(float)"},
+            "Output type is determined by the input and the specified attributes.")
+        .Attr(
+            "classes_strings",
+            "A list of strings. One and only one of classes_* should be set.",
+            AttributeProto::STRINGS,
+            OPTIONAL)
+        .Attr(
+            "classes_int64s",
+            "A list of ints.",
+            AttributeProto::INTS,
+            OPTIONAL)
+        .Attr(
+            "classes_floats",
+            "A list of floats.",
+            AttributeProto::FLOATS,
+            OPTIONAL)
+        .Attr(
+            "default_string",
+            "A string.",
+            AttributeProto::STRING,
+            std::string("\"\", empty string"))
```

Comment:
do you want to reuse the default string value as in op set 1? "_Unused". Which looks to me better.

Reply:
Sure.